src/kv/store.h

```diff
@@ -263,10 +263,14 @@ namespace kv
         }
       }
 
-      auto e = get_encryptor();
-      KvStoreSerialiser serialiser(e, v, true);
+      return snapshot;
+    }
 
-      return snapshot.serialise(serialiser);
+    std::vector<uint8_t> serialise_snapshot(
+      std::unique_ptr<AbstractSnapshot> snapshot) override
+    {
+      auto e = get_encryptor();
```

Comment:
Let's check + throw if nullptr.

Reply:
As for append entries, the serialiser still works if the encryptor is `nullptr`, i.e. if there are only public tables and the encryptor is `nullptr`, the serialisation will work. I believe this is useful to avoid boilerplate in many of our unit tests and to keep the KV and encryptor separate.

Reply:
Fair enough!